sdk/textanalytics/azure-ai-textanalytics/src/main/java/com/azure/ai/textanalytics/TextAnalyticsAsyncClient.java

```diff
@@ -0,0 +1,1371 @@
+import com.azure.ai.textanalytics.models.TextDocumentBatchStatistics;
+import com.azure.ai.textanalytics.models.TextDocumentInput;
+import com.azure.ai.textanalytics.models.TextDocumentStatistics;
+import com.azure.ai.textanalytics.models.TextSentiment;
+import com.azure.ai.textanalytics.models.TextSentimentClass;
+import com.azure.core.annotation.ReturnType;
+import com.azure.core.annotation.ServiceClient;
+import com.azure.core.annotation.ServiceMethod;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.core.util.Context;
+import com.azure.core.util.FluxUtil;
+import com.azure.core.util.logging.ClientLogger;
+import reactor.core.publisher.Mono;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import static com.azure.core.util.FluxUtil.monoError;
+import static com.azure.core.util.FluxUtil.withContext;
+
+/**
+ * This class provides an asynchronous client that contains all the operations that apply to Azure Text Analytics.
+ * Operations allowed by the client are language detection, sentiment analysis, and recognition entities, PII entities,
+ * and linked entities of a text input or list of test inputs.
+ *
+ * <p><strong>Instantiating an asynchronous Text Analytics Client</strong></p>
+ * {@codesnippet com.azure.ai.textanalytics.TextAnalyticsAsyncClient.instantiation}
+ *
+ * <p>View {@link TextAnalyticsClientBuilder} for additional ways to construct the client.</p>
+ *
+ * @see TextAnalyticsClientBuilder
+ */
+@ServiceClient(builder = TextAnalyticsClientBuilder.class, isAsync = true)
+public final class TextAnalyticsAsyncClient {
+    private final ClientLogger logger = new ClientLogger(TextAnalyticsAsyncClient.class);
+    private final TextAnalyticsClientImpl service;
+    private final TextAnalyticsServiceVersion serviceVersion;
+    private final String defaultCountryHint;
+    private final String defaultLanguage;
+
+    /**
+     * Create a {@code TextAnalyticsAsyncClient} that sends requests to the Text Analytics services's endpoint. Each
+     * service call goes through the {@link TextAnalyticsClientBuilder#pipeline http pipeline}.
+     *
+     * @param service The proxy service used to perform REST calls.
+     * @param serviceVersion The versions of Azure Text Analytics supported by this client library.
+     * @param clientOptions The {@link TextAnalyticsClientOptions client option} contains
+     * {@link TextAnalyticsClientOptions#getDefaultLanguage default language} and
+     * {@link TextAnalyticsClientOptions#getDefaultCountryHint()} default country hint} that could be used as default
+     * values for each request.
+     */
+    TextAnalyticsAsyncClient(TextAnalyticsClientImpl service, TextAnalyticsServiceVersion serviceVersion,
```

Comment:
Can you create an issue to split this monolithic class into specific package-private analytics area clients? This class has so many different responsibilities.

Reply:
Issue created - https://github.com/Azure/azure-sdk-for-java/issues/6975